miden/src/cli/run.rs

```diff
@@ -104,27 +118,83 @@ impl RunCmd {
     }
 }
 
-// HELPER FUNCTIONS
-// ================================================================================================
-
 #[instrument(name = "run_program", skip_all)]
 fn run_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Report> {
+    let ext = params
+        .program_file
+        .extension()
+        .and_then(|s| s.to_str())
+        .unwrap_or("")
+        .to_lowercase();
+
+    match ext.as_str() {
+        "masp" => run_masp_program(params),
+        "masm" => run_masm_program(params),
+        _ => Err(Report::msg("The provided file must have a .masm or .masp extension")),
+    }
+}
+
+fn run_masp_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Report> {
+    let bytes = fs::read(&params.program_file)
+        .into_diagnostic()
+        .wrap_err("Failed to read package file")?;
+
+    let package = Package::read_from_bytes(&bytes)
+        .into_diagnostic()
+        .wrap_err("Failed to deserialize package")?;
+
+    let program_arc: Arc<vm_core::Program> = match package.into_mast_artifact() {
+        MastArtifact::Executable(program_arc) => program_arc,
+        _ => return Err(Report::msg("The provided package is not a program package.")),
+    };
+    let program = &*program_arc;
+
+    let input_data: InputFile = if let Some(ref input_file_path) = params.input_file {
+        let input_data = fs::read_to_string(input_file_path)
+            .into_diagnostic()
+            .wrap_err("Failed to read input file")?;
+        serde_json::from_str(&input_data)
+            .into_diagnostic()
+            .wrap_err("Failed to deserialize input data")?
+    } else {
+        InputFile::default()
+    };
```

Comment:
Could we not do just `let input_data = InputFile::read(&params.input_file, &params.program_file)?;` instead of this?